system-tests/test_same_id_multiple_modules.py

```diff
@@ -0,0 +1,28 @@
+from helpers.kafkahelpers import create_producer, send_writer_command
+from helpers.nexushelpers import OpenNexusFileWhenAvailable
+from time import sleep
+import numpy as np
+import pytest
+
+
+@pytest.mark.skip(reason="Is painfully slow to run")
+def test_data_reaches_file(docker_compose):
```

Comment:
These are not unit tests. I suggest modifying the existing test in `test_filewriter.py rather than creating a new test.

Reply:
I can do that but I would like you to give me a rule of thumb for when to put the system tests in a separate file given that the files _test_filewriter_lr.py_, _test_filewriter.py_, _test_filewriter_ignores_commands.py_ and _test_filewriter_static_data.py_ only have one test each.

Reply:
I'm not suggesting moving your test to the existing file, I'm suggesting you modify the command for the existing test so that it also covers what your new test covers.
Broadly different scenarios that would require starting the file writer with different configurations warrant a new test, otherwise modifying an existing test is usually best. We can't have a separate system test for every unit of functionality, that is what unit tests are for.

Reply:
Ah, I understand what you mean now. This test tests a slightly different code path than the one in `test_filewriter.py`. I could in principle modify that test to also execute this code path but I am not going to because how painful it currently is to:

1. Write/modify/test system tests.
2. Modify/add the data that is pushed to the kafka broker.

Given your last comment, I see four reasonable alternatives here:

1. Do nothing.
2. Delete _test_same_id_multiple_modules.py_.
3. Delete _test_filewriter.py_.
4. Have someone other than me (e.g. you) combine the two tests.

Reply:
By the way, I am barely ok with option number 2 as I now know (because of the system test) that the code works. However, the code path executed in this case is somewhat important and should really be tested in case of future regression.